android/app/src/main/java/app/organicmaps/sdk/sync/Syncer.java

```diff
@@ -0,0 +1,354 @@
+package app.organicmaps.sdk.sync;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import app.organicmaps.sdk.bookmarks.data.BookmarkManager;
+import app.organicmaps.sdk.util.FileUtils;
+import app.organicmaps.sdk.util.concurrency.UiThread;
+import java.io.File;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class Syncer
+{
+  private final SyncClient mSyncClient;
+  private final long mAccountId; // Each account should have only one active syncer
+  private final SyncState mLocalState;
+  @NonNull
+  private final Set<String> mChangedFiles; // locally changed files
+
+  public Syncer(Context context, SyncAccount account)
+  {
+    mSyncClient = account.createSyncClient();
+    mAccountId = account.getAccountId();
+    mLocalState = new SyncState(context, account.getAccountId(), mSyncClient::computeLocalFileChecksum);
+
+    Set<String> changedFilesPref = mLocalState.getChangedFiles();
+    mChangedFiles = ConcurrentHashMap.newKeySet(changedFilesPref.size());
+    mChangedFiles.addAll(changedFilesPref);
+  }
+
+  public void markFileChanged(String filePath)
+  {
+    if (mChangedFiles.add(filePath))
+      mLocalState.setChangedFiles(mChangedFiles);
+  }
+
+  public void unmarkFileChanged(String filePath)
+  {
+    if (mChangedFiles.remove(filePath))
+      mLocalState.setChangedFiles(mChangedFiles);
+  }
+
+  public void performSync(File tempDir) throws SyncOpException, LockAlreadyHeldException
```

Comment:
<img width="300"  alt="image" src="https://github.com/user-attachments/assets/967f3ea9-295b-421c-81a5-4f131c3d70cd" />

Please, split it :)